sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -318,11 +315,56 @@ private static void validateTaskRequirements(List<TaskRequirement> taskRequireme
             }
         }
 
-        return resources;
+        return coalescePorts(resources);
+    }
+
+    private static List<Protos.Resource> coalescePorts(Collection<Protos.Resource> resources) {
+        // Within the SDK, each port is handled as its own resource, since they can have extra meta-data attached, but
+        // we can't have multiple "ports" resources on a task info, so we combine them here. Since ports are also added
```

Comment:
I don't understand.  This function still returns a list.  Was the intent to generate a single `Resource`?  Also there's a `mergeRanges` in `RangeAlgorithms.java` method for merging ports, might be useful.